lib/Listener/ShareCreatedEventListener.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace OCA\IonosProcesses\Listener;
+
+use OCA\IonosProcesses\Service\IonosMailerService;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Files\NotFoundException;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\IUserManager;
+use OCP\Share\Events\ShareCreatedEvent;
+use OCP\Share\IShare;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Notify customers about created shares via internal mail service.
+ */
+class ShareCreatedEventListener implements IEventListener {
+	public const EVENT_NAME_SHARE_BY_LINK = 'share-by-link';
+
+	public function __construct(
+		private readonly LoggerInterface $logger,
+		private readonly IUserManager $userManager,
+		private readonly IL10N $l10n,
+		private readonly IURLGenerator $urlGenerator,
+		private readonly IonosMailerService $mailer,
+	) {
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof ShareCreatedEvent)) {
+			return;
+		}
+
+		$share = $event->getShare();
+		$shareType = $share->getShareType();
+
+		if ($shareType !== IShare::TYPE_EMAIL) {
```

Comment:
this works only with `sharebymail` app enabled. as for now we disable it via https://github.com/IONOS-Productivity/easystorage-config/blob/203078d55c46befb09149491c221a6f1fff4908d/disabled-apps.inc.sh#L16

Reply:
Config update added in that submodule.